Backport ts changes

EricPedley opened the pull request:
## Summary
Merged main, then converted HelpBox to TS and fixed a bug in the recruitment banner. In resolving merge conflicts I changed a line
`flexDirection: 'left'` to `flexDirection:'row'`, which controls the layout of the recruitment banner. If you change it to column it puts the close button in a weird place so I think I've captured the original intent there. The TS compiler was complaining that 'left' wasn't a valid value for that CSS property.

This is a child PR of #411 
## Test Plan
Just make sure it builds correctly in staging and look at the recruitment banner functionality and the helpbox.

EricPedley committed:
Add link to backend repo

falsidge committed:
Fix rightpanestore (#417)

* fix calling updateFormValue statically from termSelector

* fix: Make RightPaneStore static

dependabot[bot] committed:
Bump eventsource from 1.1.0 to 1.1.2 (#409)

Bumps [eventsource](https://github.com/EventSource/eventsource) from 1.1.0 to 1.1.2.
- [Release notes](https://github.com/EventSource/eventsource/releases)
- [Changelog](https://github.com/EventSource/eventsource/blob/master/HISTORY.md)
- [Commits](https://github.com/EventSource/eventsource/compare/v1.1.0...v1.1.2)

---
updated-dependencies:
- dependency-name: eventsource
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ChaseC99 committed:
Add helpbox (#419)

* Add helpbox

* Fix 'noopener noreferrer' warning

* Fix typo

* Make text smaller

* add helpbox images

* update helpbox wording

* reduce heading margin

* Reduce size of images

Co-authored-by: Alexander Liu <git@alexanderliu.com>

EricPedley committed:
Fix RightPaneStore bugs without making everything static (#418)

EricPedley committed:
Add recruitment banner (#427)

MinhxNguyen7 committed:
Dismiss button for banner (#430)

* add recruitment banner

* Fix typo in recruitment banner

Co-authored-by: Eddy Chen <89349085+ecxyzzy@users.noreply.github.com>

* right-align banner and rewrite message

* Tooltip for Import button

* Dismiss button for banner

* Dismiss button for banner

* Recruitment banner border

* change "positions" to "opportunities"

* Extracted styles and moved bannerContainer div

* recruitment banner 11-week dismissal

* Revert "Tooltip for Import button"

Change snuck in here by accident.

* Change 6652800000 ms to representative expression

11 * 7 * 24 * 3600 * 1000 = 6652800000 = ms in a week.

Co-authored-by: Eric Pedley <ericpedley@gmail.com>
Co-authored-by: Eddy Chen <89349085+ecxyzzy@users.noreply.github.com>

MinhxNguyen7 committed:
Fix Kanban link (#436)

EricPedley committed:
Update README.md

Voark committed:
Staging Action Improvements (#433)

* Add action trigger on label/unlabel

This makes it so adding the "no deploy" label will destroy the stack if it is previously destroyed and deploys the stack if the "no deploy" label is removed.

* Change from npx cdk to aws to destroy stack

Instead of building the cdk and using npx cdk to destroy the stack we can just use the aws command which simplifies many of the build steps.

* Replace Github deployment with URL comment

Instead of dealing with the permission difficulties that come with Github deployments we instead will just use an action that comments the staging URL, and recreates on update.

* Hard code AWS credentials

Instead of using OIDC to provide temporary credentials we will use AWS credentials stored in the repo secrets. With OIDC, there will not be a good way to get AWS credentials when pull requests come from outside the repo.

* Add concurrency and wait for stack status

Because there is no reason for multiple instances of the action to run at once we add concurrency to force only one instance at a time. That being said, when cancelling another workflow instance sometimes the stack may not have finished updating so adding stack status checks ensures the action will wait until the stack is done updating before attempting more changes.

EricPedley committed:
changed 'positions' to 'opportunities' in recruitment banner (#441)

EricPedley committed:
add 2023 winter to term data

ChaseC99 committed:
Fix month indexing for Winter '23 (#445)

* Fix month indexing for Winter '23

Months are 0-indexed, so January should `0` instead of `1`

* Update src/termData.js

EricPedley committed:
fix event in too many schedules bug (#446)

EricPedley committed:
Add contributor video tutorial link

EricPedley committed:
Merge branch 'main' into backport-ts-changes

EricPedley committed:
convert HelpBox to tsx

EricPedley committed:
fix recruitment banner not being displayed

EricPedley committed:
merge from main again

Voark committed:
Fix Github Actions not checking status properly (#447)

* Fix actions stack status query

* Move stabilize stack to reusable action

* add staging environmental variable

EricPedley committed:
Merge branch 'main' into backport-ts-changes

EricPedley committed:
Update action.yaml (#451)

MinhxNguyen7 reviewed (commented):
You'd probably want the images to scale down instead of scroll right?

![image](https://user-images.githubusercontent.com/64875104/204200273-ee26c0fe-e284-49e7-b985-abc7992188ee.png)

EricPedley committed:
Merge branch 'main' into backport-ts-changes

EricPedley committed:
Fix duplicate section code bug (#453)

* make getCourseData accumulator work for duplicate section codes

* remove proxy URL

* add separate error case for 404s on user ID

ryanmohta committed:
Fix broken review links involving special characters (#455)

EricPedley committed:
Merge bugfixes from main
- section code duplicate bug
- 404 error code checking
- removing space from department codes for peterportal links

EricPedley committed:
fix bug with loading schedules
- moved ShortCourseInfo interace to AppStore
- updated UserData interface to use ShortCourseInfo instead of AppStoreCourse
- add explicit return type to getCoursesData and updated internal logic accordingly

Voark commented on src/helpers.ts:
```diff
@@ -37,32 +38,51 @@ export async function queryGraphQL(queryString: string): Promise<GradesGraphQLRe
+        prerequisiteLink: string;
+        color: string;
+        term: string;
+        scheduleIndices: number[];
```
Is there a reason this isn't just `addedCourses: AppStoreCourse[]`?

EricPedley replied:
Not really. Resolved in https://github.com/icssc/AntAlmanac/pull/450/commits/05da393a3f56fb39b4060b7dc22085d80e6c51dc

Voark commented on .github/workflows/stage_pr.yaml:
```diff
@@ -2,38 +2,31 @@ name: "Deploy Pull Request"
 
 on: 
   pull_request:
```
Looking back at this and Avent's recent PR I remember changing this line (at least locally) to [`pull_request_target`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target) which should allow outside contributors to use the action as if they made a pull request from within the repo. However this comes with the security concern of giving secret access to anyone that is able to run an action through pull requests. While the AWS role that is currently stored in the secrets should have pretty limited account access it still is a concern. Wondering if the cons are worth the benefit.

EricPedley replied:
I think we should defer making this decision to another PR since it's not directly relevant to the TS changes. This PR is just for squashing merge conflicts with main.

Voark replied:
Agreed, just remembered again when I read the code

EricPedley committed:
remove duplicate anonymous interface
